src/main/java/net/imagini/zkcli/ZkCli.java

```diff
@@ -96,99 +72,25 @@ private void doCliActions() {
         }
     }
 
-    private Optional<CuratorFramework> getCluster(CliParameters cliParameters) {
-        try {
-            return Optional.ofNullable(cliParameters.cluster == null || cliParameters.cluster.isEmpty()
-                    ? ZkClusterManager.buildClient(cliParameters.zkConnect)
-                    : ZkClusterManager.getClient(cliParameters.cluster));
-        } catch (RuntimeException ex) {
-            return Optional.empty();
-        }
-    }
-
-    private void printPathData(CuratorFramework client, String path) {
-        try {
-            System.out.println(new String(client.getData().forPath(path), CHARSET));
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
-
-    private void printPathMetaData(CuratorFramework client, String path, String metadataItemGetterName) {
-        try {
-            System.out.println(formatMetaData(client.checkExists().forPath(path), metadataItemGetterName));
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+    private void runCliActionsForPath(CuratorFramework client, String path) {
+        if (params.isListChildren()) {
```

Comment:
One thing that may help reduce a long chain of conditionals for handling actions is to define an abstract class or enum and subclass that for each unique action. That way, you can just call `actionable.doAction()` and that will delegate to the subclass' implementation of `doAction()`. I wanted to refactor this in my original PR but didn't want to bloat the scope of it too much.